siliconcompiler/scheduler.py

```diff
@@ -3,68 +3,71 @@
 import subprocess
 import stat
 import time
+import uuid
+import json
 
 ###########################################################################
 def _deferstep(chip, step, index, status):
     '''
     Helper method to run an individual step on a slurm cluster.
-    If a base64-encoded 'decrypt_key' is set in the Chip's status
-    dictionary, the job's data is assumed to be encrypted,
-    and a more complex command is assembled to ensure that the data
-    is only decrypted temporarily in the compute node's local storage.
     '''
 
     # Determine which HPC job scheduler being used.
     scheduler_type = chip.get('option', 'scheduler', 'name', step=step, index=index)
 
+    if scheduler_type != 'slurm':
+        raise ValueError(f'{scheduler_type} is not a supported scheduler')
+
     # Determine which cluster parititon to use. (Default value can be overridden on per-step basis)
     partition = chip.get('option', 'scheduler', 'queue', step=step, index=index)
+    if not partition:
+        partition = _get_slurm_partitions()
```

Comment:
Should the method name be singular instead of plural, since it only returns the first listed partition?

Reply:
Good point